schemes/musica/musica_ccpp.F90

```diff
@@ -111,20 +111,7 @@ subroutine musica_ccpp_run(time_step, temperature, pressure, dry_air_density, co
                                number_of_rate_parameters)    :: rate_parameters ! various units
     integer :: i_elem
 
-    ! Calculate photolysis rate constants using TUV-x
-    call tuvx_run(temperature, dry_air_density,                  &
-                  geopotential_height_wrt_surface_at_midpoint,   &
-                  geopotential_height_wrt_surface_at_interface,  &
-                  surface_geopotential, surface_temperature,     &
-                  surface_albedo,                                &
-                  number_of_photolysis_wavelength_grid_sections, &
-                  photolysis_wavelength_grid_interfaces,         &
-                  extraterrestrial_flux,                         &
-                  standard_gravitational_acceleration,           &
-                  cloud_area_fraction, constituents,             &
-                  air_pressure_thickness, rate_parameters,       &
-                  errmsg, errcode)
-
+    ! TODO(jiwon) this might not be correct because it doesn't know the index
```

Comment:
Hi @peverwhee, is this code still valid? We discussed that the sequence of the constituents array is not guaranteed. Do we now need to search for the index of the MICM constituents before attempting to retrieve the molar mass?
```fortran
    do i_elem = 1, size(molar_mass_arr)
      call constituent_props(i_elem)%molar_mass(molar_mass_arr(i_elem), errcode, errmsg)
      if (errcode /= 0) then
        errmsg = "[MUSICA Error] Unable to get molar mass."
        return
      end if
    end do
```

Reply:
hi @boulderdaze sorry for the delay!

Yes, you're correct that the order is not guaranteed. So you'll have to grab the index by the standard name. Haipeng created `ccpp_const_get_idx` in https://github.com/ESCOMP/atmospheric_physics/blob/main/to_be_ccppized/ccpp_const_utils.F90 for this purpose.

Reply:
Great, thank you for confirming!